backend/v3/api/instance/v2/instance.go

```diff
@@ -0,0 +1,100 @@
+package instancev2
+
+import (
+	"context"
+	"errors"
+
+	"connectrpc.com/connect"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/zitadel/zitadel/backend/v3/api/instance/v2/convert"
+	"github.com/zitadel/zitadel/backend/v3/domain"
+	"github.com/zitadel/zitadel/backend/v3/storage/database"
+	"github.com/zitadel/zitadel/backend/v3/storage/database/repository"
+	"github.com/zitadel/zitadel/internal/zerrors"
+	filter "github.com/zitadel/zitadel/pkg/grpc/filter/v2beta"
+	instance "github.com/zitadel/zitadel/pkg/grpc/instance/v2beta"
+)
+
+func DeleteInstance(ctx context.Context, instanceID string) (*connect.Response[instance.DeleteInstanceResponse], error) {
+	instanceDeleteCmd := domain.NewDeleteInstanceCommand(instanceID)
+
+	err := domain.Invoke(ctx, instanceDeleteCmd, domain.WithInstanceRepo(repository.InstanceRepository()))
+
+	if err != nil {
+		if errors.Is(err, &database.NoRowFoundError{}) {
+			return nil, zerrors.ThrowNotFound(err, "INST-QVrUwc", "instance not found")
+		}
+		return nil, err
+	}
+
+	return &connect.Response[instance.DeleteInstanceResponse]{
+		Msg: &instance.DeleteInstanceResponse{
+			// TODO(IAM-Marco): Change this with the real update date when OrganizationRepo.Update()
+			// returns the timestamp
+			DeletionDate: timestamppb.Now(),
+		},
+	}, nil
+}
+
+func GetInstance(ctx context.Context, instanceID string) (*connect.Response[instance.GetInstanceResponse], error) {
```

Comment:
IMO it should implement the `instanceconnect.InstanceServiceHandler` and therefor the request should be `*connect.Request[instance.GetInstancesRequest]`